sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -392,7 +392,9 @@ case class InsertIntoHiveTable(
 
     // Invalidate the cache.
     sqlContext.sharedState.cacheManager.invalidateCache(table)
-    sqlContext.sessionState.catalog.refreshTable(table.catalogTable.identifier)
+    if (partition.nonEmpty) {
+      sqlContext.sessionState.catalog.refreshTable(table.catalogTable.identifier)
+    }
```

Comment:
I think the reason why we call `refreshTable` here is not related to the file status cache. We are refreshing the cache because the metadata cache is out of dated.